File: app/src/main/java/br/com/example/meuprimeiroexemplo/bootstrap/APIClient.java
package br.com.example.meuprimeiroexemplo.bootstrap;

import android.widget.Toast;

import org.jetbrains.annotations.Nullable;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Passo 3 - Criar classe para mapear ENDPOINT e configurar classe para fazer o PARSER.
 * Converter JSON para objeto.
 **/
public class APIClient {

    //No retrofit você precisa colocar o  / (slash).
    //Prezado, por gentileza, informar fim de instrução (/) no endereço informado.
    private static final String ENDPOINT = "https://jsonplaceholder.typicode.com/";

    @Nullable
    public static Retrofit getClient() {
        try {
            return new Retrofit.Builder()
                    .baseUrl(ENDPOINT)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        } catch (RuntimeException e) {
            Toast.makeText(null, "deu ruimno APICLiente", Toast.LENGTH_LONG).show();
            return null;
        }
    }
}

File: app/src/main/java/br/com/example/meuprimeiroexemplo/bootstrap/APIClient.kt
@file:Suppress("KDocMissingDocumentation")

package br.com.example.meuprimeiroexemplo.bootstrap

import android.util.Log
import android.widget.Toast
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Passo 3 - Criar classe para mapear ENDPOINT e configurar classe para fazer o PARSER.
 * Converter JSON para objeto.
 */
object APIClient {
    //No retrofit você precisa colocar o  / (slash).
    //Prezado, por gentileza, informar fim de instrução (/) no endereço informado.
    private const val ENDPOINT = "https://jsonplaceholder.typicode.com/"
    val client: Retrofit?
        get() = try {
            Retrofit.Builder()
                .baseUrl(ENDPOINT)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        } catch (e: RuntimeException) {
            Toast.makeText(null, "deu ruim no $APIClient\n${e.message}", Toast.LENGTH_LONG).show()
            Log.e("APICliente", "Erro no $APIClient\n${e.message}")
            null
        }
}